OpenRA.Mods.Common/Lint/CheckPalettes.cs

```diff
@@ -75,9 +120,15 @@ static IEnumerable<string> GetPalettes(Action<string> emitError, Ruleset rules)
 					var fields = traitInfo.GetType().GetFields();
 					foreach (var field in fields.Where(x => x.HasAttribute<PaletteDefinitionAttribute>()))
 					{
+						var paletteDefinition = field.GetCustomAttributes<PaletteDefinitionAttribute>(true).FirstOrDefault();
```

Comment:
`.First()` please. the `.Where()` just above ensures that there is one, and the `if` below will stop complaining about a "Possible null reference exception".